File: 318-maximum-product-of-word-lengths/318-maximum-product-of-word-lengths.java
class Solution {
    public int maxProduct(String[] words) {
         int len = words.length;
    int[] set = new int[len];
    for (int i = 0; i < len; i++) {
      for (char a : words[i].toCharArray()) {
        set[i] |= (1 << a - 'a');
      }
    }
    int max = 0;
    for (int i = 0; i < len; i++) {
      for (int j = i + 1; j < len; j++) {
        if ((set[i] & set[j]) == 0) {
            max = Math.max(words[i].length() * words[j].length(), max);
        }
      }
    }
    return max;
    }
}

File: 318-maximum-product-of-word-lengths/318-maximum-product-of-word-lengths.cpp
class Solution {
public:
    int maxProduct(vector<string>& words) {
        vector<int> value(words.size(), 0);
        for(int i = 0; i < words.size(); ++i)
            for(auto c: words[i])
                value[i] |= 1 << (c - 'a');

        int ret = 0;
        for(int i = 1; i < words.size(); ++i)
            for(int j = 0; j < i; ++j)
                if((value[i] & value[j]) == 0 && words[i].length() * words[j].length() > ret)
                    ret = words[i].length() * words[j].length();
        return ret; 
    }
};